LocaleEmulator/Hooks/NtdllHook.cpp

```diff
@@ -872,6 +872,27 @@ LONG NTAPI LeKnownExceptionFilter(PEXCEPTION_POINTERS ExceptionPointers)
     return Result;
 }
 
+NTSTATUS NTAPI LeCustomCPToUnicodeN(IN PCPTABLEINFO CustomCP,
+    OUT PWCHAR UnicodeString,
+    IN ULONG UnicodeSize,
+    OUT PULONG ResultSize OPTIONAL,
+    IN PCHAR CustomString,
+    IN ULONG CustomSize) {
+    PLeGlobalData GlobalData = LeGetGlobalData();
+
+    if (CustomCP->CodePage != 65001 && CustomCP->CodePage != GlobalData->GetLeb()->AnsiCodePage) {
+        PROTECT_SECTION(&GlobalData->HookRoutineData.Ntdll.NtLock) {
+            if (CustomCP->CodePage != 65001 && CustomCP->CodePage != GlobalData->GetLeb()->AnsiCodePage) {
```

Comment:
double `if` checks

Reply:
This double check is meant to prevent some write race, just imagine if there are concurrently two threads invoking `RtlCustomCPToUnicodeN()`. But if you insist to remove the double check, I will do so.

Reply:
I don't think that happens but yeah, it can be kept as-is.